src/olympia/shelves/views.py

```diff
@@ -0,0 +1,12 @@
+from rest_framework import viewsets
+
+from olympia.shelves.models import Shelf
+from olympia.shelves.serializers import ShelfSerializer
+
+
+class ShelfViewSet(viewsets.ModelViewSet):
+    queryset = Shelf.objects.filter(
+        shelfmanagement__enabled=True).order_by('shelfmanagement__position')
+    permission_classes = []
+
+    serializer_class = ShelfSerializer
```

Comment:
can we have a test for the view too?  Doesn't need to do anything complicated - just check that it does indeed return a couple of Shelf instances, in the correct order, and without disabled shelves.

Reply:
I added a `test_views.py`, but I'm not 100% sure about it. I got a `NoReverseMatch` error stating `Reverse for 'addon-search' not found. 'addon-search' is not a valid view function or pattern name` when I tried to use `ShelfSerializer(instance=shelf_a).data`, so I wrote lines 64-68 instead. Please take a look when you have a moment.

Reply:
yes, that's fine - ShelfSerializer now relies on a request being passed in to resolve the api url correctly so `ShelfSerializer(instance=shelf_a).data` won't work